src/Orleans/Messaging/ProxiedMessageCenter.cs

```diff
@@ -296,16 +307,26 @@ public Message WaitMessage(Message.Categories type, CancellationToken ct)
 #endif
                 return msg;
             }
-            catch (ThreadAbortException tae)
+            catch (ThreadAbortException exc)
             {
                 // Silo may be shutting-down, so downgrade to verbose log
-                logger.Verbose(ErrorCode.ProxyClient_ThreadAbort, "Received thread abort exception -- exiting. {0}", tae);
+                logger.Verbose(ErrorCode.ProxyClient_ThreadAbort, "Received thread abort exception -- exiting. {0}", exc);
                 Thread.ResetAbort();
                 return null;
             }
-            catch (OperationCanceledException oce)
+            catch (OperationCanceledException exc)
+            {
+                logger.Verbose(ErrorCode.ProxyClient_OperationCancelled, "Received operation cancelled exception -- exiting. {0}", exc);
+                return null;
+            }
+            catch (ObjectDisposedException exc)
+            {
+                logger.Verbose(ErrorCode.ProxyClient_ReceiveError, "Received Object Disposed exception -- exiting. {0}", exc);
+                return null;
+            }
+            catch (InvalidOperationException exc)
             {
-                logger.Verbose(ErrorCode.ProxyClient_OperationCancelled, "Received operation cancelled exception -- exiting. {0}", oce);
+                logger.Verbose(ErrorCode.ProxyClient_ReceiveError, "Received Invalid Operation exception -- exiting. {0}", exc);
```

Comment:
yes, I agree. Go ahead.
Also, those are at Verbose, so they wont appear in the log at all, since we very rarely run at Verbsoe.